### PR TITLE
Add documentation of `elementId` function

### PR DESCRIPTION
**Pull request by loveleif**
- [CIP](https://docs.google.com/document/d/1TzfAeGOrsLqti-6pG5l1w8tE4wY0_Kush9Ko3OAjYfQ/edit)
- [Feature PR](https://github.com/neo-technology/neo4j/pull/14823)

### CODE REVIEW

**Comment by loveleif** [comment]
[Documentation for elementId function](https://trello.com/c/uuIiR8nr/3377-documentation-for-elementid-function)

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 1]
```diff
@@ -87,6 +88,42 @@ class ScalarFunctionsTest extends DocumentingTest {
...
+            #* Every node and relationship has an element id.
+            #* The identifier is unique among other nodes' and relationships' identifiers within the scope of a single transaction and DBMS.
+            #  In other words, element id is also unique across different databases/graphs in the same DBMS.
+            #
+            #Example of properties that are NOT guaranteed:
+            #
+            #* No guarantees are given regarding the order of the returned id values.
+            #* No guarantees are given about the length of the id string values.
+            #* Outside of the scope of a single transaction, no guarantees are given about the mapping between id values and elements.
+          """.stripMargin('#'))
```
Comment (loveleif) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala]: Suggestions for language improvements are very welcome, had a hard time expressing this. Thanks!

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 2]
```diff
@@ -87,6 +88,42 @@ class ScalarFunctionsTest extends DocumentingTest {
...
       }
     }
+    section("elementId()", "functions-element-id") {
+      p("""The function `elementId()` returns a node or relationship identifier, unique within a specific transaction and DBMS.
+        """.stripMargin('#'))
+      note {
+        p("""Guarantees:
+            #
+            #* Every node and relationship has an element id.
+            #* The identifier is unique among other nodes' and relationships' identifiers within the scope of a single transaction and DBMS.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala]:
```suggestion
         Every node and relationship is guaranteed an element id.
         This id is unique among both nodes and relationships across all databases in the same DBMS within the scope of a single transaction.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 3]
```diff
@@ -87,6 +88,42 @@ class ScalarFunctionsTest extends DocumentingTest {
         resultTable()
       }
     }
+    section("elementId()", "functions-element-id") {
+      p("""The function `elementId()` returns a node or relationship identifier, unique within a specific transaction and DBMS.
+        """.stripMargin('#'))
+      note {
+        p("""Guarantees:
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala]:
```suggestion
        p("""
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 4]
```diff
@@ -87,6 +88,42 @@ class ScalarFunctionsTest extends DocumentingTest {
...
+            #
+            #* Every node and relationship has an element id.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala]:
```suggestion
            
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 5]
```diff
@@ -87,6 +88,42 @@ class ScalarFunctionsTest extends DocumentingTest {
...
+            #* The identifier is unique among other nodes' and relationships' identifiers within the scope of a single transaction and DBMS.
+            #  In other words, element id is also unique across different databases/graphs in the same DBMS.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala]:
```suggestion

```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 6]
```diff
@@ -87,6 +88,42 @@ class ScalarFunctionsTest extends DocumentingTest {
...
+            #
+            #Example of properties that are NOT guaranteed:
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala]:
```suggestion
            # However, no guarantees are given regarding the order of the returned id values or the length of the id string values.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 7]
```diff
@@ -87,6 +88,42 @@ class ScalarFunctionsTest extends DocumentingTest {
...
+            #
+            #* No guarantees are given regarding the order of the returned id values.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala]:
```suggestion
            
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 8]
```diff
@@ -87,6 +88,42 @@ class ScalarFunctionsTest extends DocumentingTest {
...
+            #* No guarantees are given about the length of the id string values.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala]:
```suggestion
          
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 9]
```diff
@@ -87,6 +88,42 @@ class ScalarFunctionsTest extends DocumentingTest {
...
+            #* Outside of the scope of a single transaction, no guarantees are given about the mapping between id values and elements.
+          """.stripMargin('#'))
+      }
+      function("elementId(expression)",
+        "A String.",
+        ("expression", "An expression that returns a node or a relationship."))
+      considerations(
+        """* `elementId(null)` returns `null`.
+          |* `elementId` on values other than a node, relationship or null will fail the query.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala]:
```suggestion
          |* `elementId` on values other than a node, relationship, or null will fail the query.
```

**Review by AlexicaWright — COMMENTED** [review]
I have not been able to build this locally as the tests are failing. Once those errors have been addressed, I'd be happy to try again. But in the meantime, I've added some suggestion for better readability.

**Comment by loveleif** [comment]
> I have not been able to build this locally as the tests are failing.

@AlexicaWright sorry about that, the feature PR is not merged yet, should have been clearer with that. Thanks for the review!! After I have merged in the mono repo I'll make sure this builds.

**Comment by loveleif** [comment]
The failures that are left will be fixed with #1454.

**Review by AlexicaWright — APPROVED** [review]
LGTM 👍 Thanks @loveleif !

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala [hunk 10]
```diff
@@ -120,7 +150,8 @@ class ScalarFunctionsTest extends DocumentingTest {
         resultTable()
       }
     }
-    section("id()", "functions-id") {
+    section("[deprecated]#id()", "functions-id") {
```
Comment (martin-neotech) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ScalarFunctionsTest.scala:153]: This delta is missing in the https://development.neo4j.dev/docs/cypher-manual/5.0/deprecations-additions-removals-compatibility/

**Pull request merged**